### PR TITLE
Document TFS and TS support in perf_analyzer

### PR DESCRIPTION
**Pull request by tanmayv25**

### CODE REVIEW

File: docs/perf_analyzer.md [hunk 1]
```diff
@@ -511,3 +511,53 @@ There are a few functionalities that are missing from the C API. They are:
...
+## Benchmarking TensorFlow Serving
+perf_analyzer can also be used to benchmark models deployed on [TensorFlow Serving](https://github.com/tensorflow/serving) using the `--service-kind` option. The support is however only available through gRPC protocol.
+ 
+Following invocation demonstrates how to configure perf_analyzer to issue requests to a running instance of tensorflow_model_server:
+ 
+```
+perf_analyzer -m <model_name> --service-kind=tfserving -i grpc -u localhost:8500
+```
+ 
+You might have to specify a different url(`-u`) to access wherever the server is running. The report of perf_analyzer will only include statistics measured at the client-side.
```
Comment (GuanLuo) [inline docs/perf_analyzer.md:525]: Should there be a snippet of the PA output?

**Review by GuanLuo — COMMENTED** [review]
Break the long paragraphs in multiple lines.

File: docs/perf_analyzer.md [hunk 2]
```diff
@@ -511,3 +511,53 @@ There are a few functionalities that are missing from the C API. They are:
...
    [qa/L0_perf_analyzer_capi/test.sh](https://github.com/triton-inference-server/server/blob/main/qa/L0_perf_analyzer_capi/test.sh#L239-L277)
+
+
+## Benchmarking TensorFlow Serving
+perf_analyzer can also be used to benchmark models deployed on [TensorFlow Serving](https://github.com/tensorflow/serving) using the `--service-kind` option. The support is however only available through gRPC protocol.
+ 
+Following invocation demonstrates how to configure perf_analyzer to issue requests to a running instance of tensorflow_model_server:
+ 
+```
+perf_analyzer -m <model_name> --service-kind=tfserving -i grpc -u localhost:8500
```
Comment (CoderHam) [inline docs/perf_analyzer.md:522]: Is the default port 8500?

File: docs/perf_analyzer.md [hunk 3]
```diff
@@ -511,3 +511,53 @@ There are a few functionalities that are missing from the C API. They are:
...
+```
+ 
+You might have to specify a different url(`-u`) to access wherever the server is running. The report of perf_analyzer will only include statistics measured at the client-side.
+ 
+**NOTE:** The support is still in a **beta** quality. perf_analyzer does not guarantee the most optimum tuning for TensorFlow Serving.
```
Comment (CoderHam) [inline docs/perf_analyzer.md:527]: Remove quality

File: docs/perf_analyzer.md [hunk 4]
```diff
@@ -511,3 +511,53 @@ There are a few functionalities that are missing from the C API. They are:
...
+2. `Different library versions`:
+The version of TensorFlow might differ between Triton and TensorFlow Serving being benchmarked. Even the versions of cuda libraries might differ between the two solutions. The performance of models can be susceptible to the versions of these libraries. For a single request concurrency, if the compute_infer time reported by perf_analyzer when benchmarking Triton is as large as the latency reported by perf_analyzer when benchmarking TFS, then the performance difference is likely because of the difference in the software stack and outside the scope of Triton.
+3. `CPU Optimization`:
+TFS has separate builds for CPU and GPU targets. They have target-specific optimization. Unlike TFS, Triton has a single build which is optimized for execution on GPUs. When collecting performance on CPU models on Triton, try running Triton with the environment variable `TF_ENABLE_ONEDNN_OPTS=1`.
+ 
+ 
+## Benchmarking TorchServe
+perf_analyzer can also be used to benchmark [TorchServe](https://github.com/pytorch/serve)  using the `--service-kind` option. The support is however only available through HTTP protocol. It also requires input to be provided within a file.
```
Comment (CoderHam) [inline docs/perf_analyzer.md:540]: Remove extra space before `using`

Comment (CoderHam) [inline docs/perf_analyzer.md:540]: input to be provided via a json file

File: docs/perf_analyzer.md [hunk 5]
```diff
@@ -511,3 +511,53 @@ There are a few functionalities that are missing from the C API. They are:
...
+ 
+Following invocation demonstrates how to configure perf_analyzer to issue requests to a running instance of torchserve assuming the location holds `kitten_small.jpg`:
```
Comment (CoderHam) [inline docs/perf_analyzer.md:542]: TorchServe

File: docs/perf_analyzer.md [hunk 6]
```diff
@@ -511,3 +511,53 @@ There are a few functionalities that are missing from the C API. They are:
...
+       {
+         "TORCHSERVE_INPUT" : ["kitten_small.jpg"]
+       }
+     ]
+ }
+```
+ 
+You might have to specify a different url(`-u`) to access wherever the server is running. The report of perf_analyzer will only include statistics measured at the client-side.
+ 
+**NOTE:** The support is still in a **beta** quality. perf_analyzer does not guarantee the most optimum tuning for TorchServe.
```
Comment (CoderHam) [inline docs/perf_analyzer.md:563]: Remove quality

Comment (CoderHam) [reply docs/perf_analyzer.md:563]: remove `the most`

**Review by CoderHam — CHANGES_REQUESTED** [review]

File: docs/perf_analyzer.md (hunk 2 above)
Comment (tanmayv25) [reply docs/perf_analyzer.md]: Yes

File: docs/perf_analyzer.md (hunk 5 above)
Comment (tanmayv25) [reply docs/perf_analyzer.md]: fixed to `torchserve`. I want to describe the binary file here.

File: docs/perf_analyzer.md [hunk 7]
```diff
@@ -537,15 +556,34 @@ TFS has separate builds for CPU and GPU targets. They have target-specific optim
  
  
 ## Benchmarking TorchServe
-perf_analyzer can also be used to benchmark [TorchServe](https://github.com/pytorch/serve)  using the `--service-kind` option. The support is however only available through HTTP protocol. It also requires input to be provided within a file.
+perf_analyzer can also be used to benchmark [TorchServe](https://github.com/pytorch/serve) using the `--service-kind` option. The support is however only available through HTTP protocol. It also requires input to be provided via a json file.
```
Comment (GuanLuo) [inline docs/perf_analyzer.md]: JSON

**Review by GuanLuo — APPROVED** [review]

**Review by CoderHam — APPROVED** [review]

File: docs/perf_analyzer.md [hunk 8]
```diff
@@ -511,3 +511,157 @@ There are a few functionalities that are missing from the C API. They are:
...
+[instance count](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#instance-groups). Hence, at a higher request concurrency, most
+of the requests are blocked on the instance availability. To
+configure Triton to behave like TFS, set the instance count to a
+reasonably high value and then set
+[MAX_SESSION_SHARE_COUNT](https://github.com/triton-inference-server/tensorflow_backend#parameters)
+parameter in the model confib.pbtxt to the same value.For some
+context, the TFS sets its thread constraint to four times the
+num of schedulable CPUs.
+2. `Different library versions`:
+The version of TensorFlow might differ between Triton and
```
Comment (matthewkotila) [inline docs/perf_analyzer.md:576]: Can the grammar of this sentence be changed? It seems hard to understand.

> ... Tensorflow Serving _when_ being benchmarked. ...

**Pull request merged**